docs/tutorials/services/color-detection-scuttle.md

```diff
@@ -17,105 +17,383 @@ languages: [ "python" ]
+3. Navigate to the **Control** tab, click on your transform camera and toggle it on.
+   The transform camera will now show detections with bounding boxes around the detected colors.
 
-## Run the code
+{{< alert title="Tip" color="tip" >}}
+If the color is not reliably detected, increase the `hue_tolerance_pct`.
 
-Now you should try to drive the SCUTTLE around following the color red.
-You can use something like a red sports ball or book cover as a target to follow.
+Note that the detector does not detect black, perfect greys (greys where the red, green, and blue color component values are equal), or white.
+{{< /alert >}}
+
+## Program your rover
 
-Navigate to the folder where you saved the Python script.
-From that folder, run this in the terminal:
+### Set up your code environment
+
+We are going to use Virtualenv to set up a virtual environment for this project, in order to isolate the dependencies of this project from other projects.
+Run the following commands in your command-line to install virtualenv, set up an environment `venv` and activate it:
 
 ```sh {class="command-line" data-prompt="$"}
-python scuttle.py
+python3 -m pip install --user virtualenv
+python3 -m venv env
+source env/bin/activate
 ```
 
-Be sure to replace <file>~/Desktop/</file> with the <file>/path/toYour/directory/</file> where the Python code was saved.
+Now, install the Python Viam SDK:
 
 ```sh {class="command-line" data-prompt="$"}
-python ~/Desktop/scuttle.py
+pip3 install viam-sdk python-vlc
 ```
 
-## Notes on color detection operation
+### Connect
+
+Next, go to the **Code sample** tab on your robot page and select **Python**, then click **Copy**.
+
+{{% snippet "show-secret.md" %}}
+
+This code snippet imports all the necessary packages and sets up a connection with the Viam app in the cloud.
 
-Within `getVisService(robot)`, a detector is configured with particular properties and subsequently added to the vision service.
-This particular detector is a [color detector](/services/vision/detection/), which means the relevant parameters are `detect_color` (hex string), `hue_tolerance_pct` (float from `0` to `1`), and `segment_size_px` (integer).
-Feel free to configure more detectors with different parameters!
-To learn about all the different detectors and parameters, check out the [vision service documentation](/services/vision/).
+Next, create a file named <file>main.py</file> and paste the boilerplate code from the **Code sample** tab of the Viam app into your file.
+Then, save your file.
 
-The `leftOrRight()` code splits the screen vertically into thirds (left, center, and right) and makes a determination about which third the object (red ball) is in.
-Within `main()`, this decides how the robot moves (as configured by the 4 given variables).
-Run the code as is before making changes to see how it affects the output!
+Run the code to verify that the Viam SDK is properly installed and that the `viam-server` instance on your robot is live.
```

Comment:
Ah, looks like a running instance of `viam-server` is also a requirement: it looks like, for _Configure color detection_, we assume an existing robot, which assumes an existing `viam-server` installation.

I think we'd be fine here just adding a link here, maybe:

```suggestion
Run the code to verify that the Viam SDK is properly installed and that the `viam-server` instance on your robot is live.
If you haven't yet installed `viam-server`, follow the [installation guide](/installation/#install-viam-server) to install `viam-server` on your robot before proceeding with this tutorial.
```